security-framework/src/os/macos/passwords.rs

```diff
@@ -0,0 +1,166 @@
+    let account_name_len = account.len() as u32;
+    let account_name = CString::new(account).unwrap();
+
+    let password_len = password.len() as u32;
+    let mut item = ptr::null_mut();
+
+    unsafe {
+        let status = SecKeychainFindGenericPassword(ptr::null(),
+                                                    service_name_len,
+                                                    service_name.as_ptr(),
+                                                    account_name_len,
+                                                    account_name.as_ptr(),
+                                                    ptr::null_mut(),
+                                                    ptr::null_mut(),
+                                                    &mut item);
+
+        match status {
+            errSecSuccess => {
+                try!(cvt(SecKeychainItemModifyAttributesAndData(
+                    item, ptr::null(), password_len, password.as_ptr())));
+            },
+            _ => {
+                try!(cvt(SecKeychainAddGenericPassword(ptr::null_mut(),
+                                                       service_name_len,
+                                                       service_name.as_ptr(),
+                                                       account_name_len,
+                                                       account_name.as_ptr(),
+                                                       password_len,
+                                                       password.as_ptr(),
+                                                       ptr::null_mut())));
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Delete a generic password.
+pub fn delete_generic_password(service: &str, account: &str) -> Result<()> {
+    let service_name_len = service.len() as u32;
+    let service_name = CString::new(service).unwrap();
+
+    let account_name_len = account.len() as u32;
+    let account_name = CString::new(account).unwrap();
+
+    let mut item = ptr::null_mut();
+
+    unsafe {
+         try!(cvt(SecKeychainFindGenericPassword(ptr::null(),
+                                                service_name_len,
+                                                service_name.as_ptr(),
+                                                account_name_len,
+                                                account_name.as_ptr(),
+                                                ptr::null_mut(),
+                                                ptr::null_mut(),
+                                                &mut item)));
+
+        SecKeychainItemDelete(item);
+        CFRelease(item);
+        Ok(())
+    }
+}
+
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn missing_password() {
+        let password = find_generic_password("this_service_does_not_exist",
+                                             "this_account_is_bogus");
+        assert!(password.is_err());
+    }
+
+    #[test]
+    fn round_trip_password() {
```

Comment:
It'd probably be nice to have these use a temporary keychain to avoid polluting the system one.